src/csharp/Microsoft.Spark/Sql/SparkSession.cs

```diff
@@ -389,6 +390,39 @@ public void Dispose()
         /// </summary>
         public void Stop() => _jvmObject.Invoke("stop");
 
+        /// <summary>
+        /// Get the <see cref="VersionSensor.VersionInfo"/> for the Microsoft.Spark assembly and
+        /// make a "best effort" attempt in determining the version of Microsoft.Spark.Worker
+        /// assembly.
+        /// </summary>
+        /// <returns>
+        /// A <see cref="DataFrame"/> containing the <see cref="VersionSensor.VersionInfo"/>
+        /// </returns>
+        public DataFrame Version()
+        {
+            DataFrame sparkInfoDf =
+                CreateDataFrame(
+                    new GenericRow[] { VersionSensor.MicrosoftSparkVersion().ToGenericRow() },
+                    VersionSensor.VersionInfo.s_schema);
+
+            Func<Column, Column> workerInfoUdf =
+                Functions.Udf<int>(
+                    i => VersionSensor.MicrosoftSparkWorkerVersion().ToGenericRow(),
+                    VersionSensor.VersionInfo.s_schema);
+            DataFrame df = CreateDataFrame(Enumerable.Range(0, 1000));
+
+            string tempColName = "WorkerVersionInfo";
+            DataFrame workerInfoTempDf = df
+                .Repartition(1000)
```

Comment:
the `1000` is too big for local settings.  Is there some spark setting we can leverage to set this value ?  Or what do you think are safe defaults ?  we can also pass this in as a parameter and have a safe default of 10 or something ?